quickwit/quickwit-serve/src/elastic_search_api/bulk.rs

```diff
@@ -78,23 +79,37 @@ pub fn es_compat_index_bulk_handler(
             elastic_ingest_bulk(Some(index), body, ingest_option, ingest_service)
         })
         .and(extract_format_from_qs())
-        .map(make_json_api_response)
+        .map(make_elastic_api_response)
+}
+
+fn make_elastic_api_response(
+    elasticsearch_result: Result<IngestResponse, ElasticSearchError>,
+    format: BodyFormat,
+) -> JsonApiResponse {
+    let status_code = match &elasticsearch_result {
+        Ok(_) => StatusCode::OK,
+        Err(err) => err.status,
+    };
+    JsonApiResponse::new(&elasticsearch_result, status_code, &format)
 }
 
 async fn elastic_ingest_bulk(
     index: Option<String>,
     body: Bytes,
     ingest_options: ElasticIngestOptions,
     mut ingest_service: IngestServiceClient,
-) -> Result<IngestResponse, IngestRestApiError> {
+) -> Result<IngestResponse, ElasticSearchError> {
     let mut doc_batch_builders = HashMap::new();
-    let mut lines = lines(&body);
+    let mut lines = lines(&body).enumerate();
 
-    while let Some(line) = lines.next() {
-        let action = serde_json::from_slice::<BulkAction>(line)
-            .map_err(|error| IngestRestApiError::BulkInvalidAction(error.to_string()))?;
-        let source = lines.next().ok_or_else(|| {
-            IngestRestApiError::BulkInvalidSource("expected source for the action".to_string())
+    while let Some((line_number, line)) = lines.next() {
+        let action = serde_json::from_slice::<BulkAction>(line).map_err(|error| {
```

Comment:
with this replacement `BulkInvalidAction`/ `BulkInvalidSource` became redundant, but I didn't remove them for now since potentially the whole `IngestRestApiError` could be removed